src/background/stores/card-selection-store.ts

```diff
@@ -24,17 +24,72 @@ export class CardSelectionStore extends BaseStoreImpl<CardSelectionStoreData> {
 
     public getDefaultState(): CardSelectionStoreData {
         const defaultValue: CardSelectionStoreData = {
-            rules: null,
+            rules: {},
         };
 
         return defaultValue;
     }
 
-    public toggleRuleExpandCollapse(payload: RuleExpandCollapsePayload): void {}
+    private deselectAllCardsInRule = (rule: RuleExpandCollapseData): void => {
+        if (!rule) {
+            return;
+        }
 
-    public toggleCardSelection(payload: CardSelectionPayload): void {}
+        for (const resultInstanceUid of Object.keys(rule.cards)) {
+            rule.cards[resultInstanceUid] = false;
+        }
+    };
 
-    public CollapseAllRules(): void {}
+    public toggleRuleExpandCollapse = (payload: RuleExpandCollapsePayload): void => {
+        if (!payload || !payload.ruleId || !this.state.rules || !this.state.rules[payload.ruleId]) {
+            return;
+        }
 
-    private onScanCompleted = (payload: UnifiedScanCompletedPayload): void => {};
+        const rule = this.state.rules[payload.ruleId];
+
+        rule.isExpanded = !rule.isExpanded;
+
+        if (!rule.isExpanded) {
+            this.deselectAllCardsInRule(rule);
+        }
+
+        super.emitChanged();
```

Comment:
The pattern we have for emit changed is to call it using "this" rather than "super"; doesn't really functionally matter (as far as I know) but I'd prefer to be consistent across our code.

Reply:
Done.